### DISCUSSION TITLE
New Feature Ideas

### DISCUSSION THREAD

**Discussion by kylecorry31** [kylecorry31/Trail-Sense#316] [New Feature Ideas]
Feel free to post new feature ideas here, or within the issue created for them

**Comment by jer194** [comment]
Hi, maybe recording of tracks would fit into the scope of this app, unless it is there already.
Also the app name 'Trail-Sense' could make you think this feature is included.
Up to now I'm mainly using osmtracker for this purpose.

And for temperatur / weather panel, there could be a similar history graph like for pressure.

**Reply by kylecorry31** [reply]
Hi, thank you for the suggestions! You will be happy to hear that a new feature that will be releasing soon, called 'Backtrack', will provide the ability to record tracks as waypoints and navigate back to them. Until map support is added through either TS Maps (https://github.com/kylecorry31/Trail-Sense-Maps) or https://github.com/kylecorry31/Trail-Sense/issues/314, you will only be able to see your tracks as waypoints (which are a lot like beacons). 

As for the temperature history, I just created https://github.com/kylecorry31/Trail-Sense/issues/394 to keep track of that - it should be really easy to add (and there's also https://github.com/kylecorry31/Trail-Sense/issues/200, though I've never had a phone with a humidity sensor, so it's harder to test that one).

**Reply by jer194** [reply]
'Backtrack': thanks, I can see it already in my own build. I'll try it soon.

'Temperature': excellent :-)
'Humidity': I can test if you want (on Galaxy S4)

**Reply by kylecorry31** [reply]
Great, thanks!

When I have the humidity history in place, I can tag you here

**Reply by fillwithjoy1** [reply]
Hey @kylecorry31
Just wondering if humidity has been added to the app yet.

**Reply by kylecorry31** [reply]
Yes, it was added in late 2022 - I completely forgot to reply to this thread when implemented. If the device has a humidity sensor (very rare, mainly older Samsungs), it should show up on the weather tab and you can click it to see the humidity history.

I have not yet added historical humidity values for devices without a hygrometer though: #1567

**Comment by qwerty287** [comment]
An idea that could resolve #347 (at least partially) would be to add quick actions that you can open if you do a long click on the app icon on the launcher, for example see the image that shows these quick actions of the system settings.
![Screenshot_1619593883](https://user-images.githubusercontent.com/80460567/116362284-2f2fe100-a802-11eb-98df-477309390e58.png)

**Reply by qwerty287** [reply]
Should I open an issue for this?

**Reply by kylecorry31** [reply]
No, that was part of the referenced issue (I have a comment on that about quick actions)

**Reply by kylecorry31** [reply]
Actually, I can create a new issue for this later - it appears there is a different amount of pre-work for each (as in, what I want the actions to be and setting up the URI and registering it for the shortcuts)

The widget issue itself has a lot more pre-work involved as I want to avoid code duplication there.

Therefore they can be completed at different points in time

**Comment by tomeckip** [comment]
Hi, I think a nice feature/tool would be to display total elevation gained/lost during a hike. It seems like this is almost implemented with the elevation graph display, but I don't believe there is any way to start recording at the beginning of a hike and have the app show the total elevation gained and lost.

As the different methods of determining elevations have their pros and cons and some phones do not have a barometric sensor, perhaps both barometric based elevation and DEM referenced from GPS location could be implemented.

**Reply by kylecorry31** [reply]
Hi, thank you for the suggestion. I believe the feature you are looking for is Backtrack. From the Paths tool, if you click the start icon at the bottom right for Backtrack, it will begin creating a path and you'll be able to see the elevation gained / lost for that (Ascent and Descent). You can adjust the waypoint frequency in Trail Sense Settings > Navigation > Backtrack interval. If that is not what you are looking for, can you elaborate further?

As for the elevation methods, Backtrack will use what is configured in the Altimeter settings (Trail Sense Settings > Sensors > Altimeter). The options are barometer or GPS elevation (not DEM). Trail Sense does not have DEM support due to the large amount of data that would be required for that (and no Internet access) - I have #1540 to investigate creating a plugin for DEM support to avoid including it in the base application.

**Reply by tomeckip** [reply]
Thank you! I hadn't explored that tool yet.

**Comment by fillwithjoy1** [comment]
Just a question, do you currently use the ambient temperature sensor when using the thermometer set to sensor mode? Thanks

**Reply by kylecorry31** [reply]
Hi, yes when the thermometer is set to sensor mode it will determine which sensor to use in the following order based on availability, preferring the ambient temperature sensor when available.

1. Ambient temperature sensor (on a select few devices, mainly Samsung)
2. Temperature sensor (not on many devices, it was the older version of the ambient temperature sensor)
3. Battery temperature sensor (supported by most devices)

**Comment by GuiSousa135** [comment]
Hello, I'm a Trail Sense user and I looove this project, a few days ago I suggested the inssue of Spherical Planetarium with gyroscope (like in the Sky Map) to add to the Astronomy tab of the app, I think it would be something amazing to be able to follow the planets and stars during a nocturnal observation or finding the position of stars to photograph. Do you have plans to bring this function in the future? And how can I help with the implementation? Ps: I'm not a programmer but a designer. Thanks for your work, it helps me a lot with night trekking, hiking and astrophotograph. 😁

**Reply by kylecorry31** [reply]
Thank you for the suggestion! I plan on adding that at some point in the future (not within the next year though). Please see my comment here: #1809 . I plan on only adding a very limited number of stars and planets to fit into Trail Sense's core use cases.

Thank you for the offer to help! I would find it very helpful if you could find other apps doing this and maybe provide suggestions on what Trail Sense should mimic or do differently from them to be easy to use. Or if you can assemble a list of stars and planets which meet the criteria I listed on my comment in #1809. 

If you can add the suggestions to #1809, that will make it easier for me to implement.

**Reply by jer194** [reply]
I'm using this occasionally, because it is small, and OpenSource:

https://github.com/tengel/AndroidPlanisphere

I agree, something like this would be nice to have in TrailSense, however,
it might make it less lightweight ... and for me it is ok to use it as an independent app.

**Reply by kylecorry31** [reply]
Thanks! That's one of the reasons I only plan on adding a subset of stars and planets - the main focus of this tool will be celestial navigation.

**Comment by jer194** [comment]
Hi again,

we just have discussed how to convert knots to km/h or m/s.
Rightly I have assumed TrailSense has units conversions included already :-),
though 'velocities' (or 'speed') is still missing there. Maybe it could be added as well?
(Beaufort would be another candidate, though it is more for wind speed ranges / damage it can cause).

Thanks for your continued efforts!!
Jakob.

**Reply by kylecorry31** [reply]
Good idea, I'll add that (https://github.com/kylecorry31/Trail-Sense/issues/1897)

**Comment by giedriusva1** [comment]
Hi,
I would like to have an option to increase font size. In most tools, there is quite some whitespace. 
To avoid needing glasses to use the app. For some older folks :)
Many thanks

**Reply by kylecorry31** [reply]
Hi, Trail Sense follows the system font size settings - can you confirm that works for you? If that isn't acceptable, I can look into having a custom setting in Trail Sense for it (and make it by default follow the system setting).

On my phone, I can access this by going to my phone's Settings > Display > Display size and text > Font size

**Comment by jer194** [comment]
Hi,
would it be possible to add the current altitude and azimuth to the details page for Moon (and Sun)?
It might be useful in certain circumstances ...
Thanks - Jakob.

**Reply by kylecorry31** [reply]
Hi, yes that is possible - I created this issue to track that: https://github.com/kylecorry31/Trail-Sense/issues/2289

For now, if you click the sun or moon images on the chart, you can see the current altitude/azimuth of both the sun and moon below the chart.

**Reply by jer194** [reply]
Thanks, good to know!
I guess I should have read the documentation :-)

**Reply by kylecorry31** [reply]
No problem, also related I'm planning on adding the azimuth of rise/set times to the details as well (#1339)

**Reply by jer194** [reply]
Thanks for your incredible dedication and agility!

**Comment by CloneWith** [comment]
Hi. Something about the Beacon and Paths tab:

```md
- Group A
   - Item A in Group A
- Item A
- Item B
```

Can we show the structure in a tree-style form? To use an 'expand' button to show items in a group.
Or consider adding an 'upper' button inside each Group? So we don't need to press the Back button every time.

---

Also, I suggest adding support for multiple items choosing (by pressing on icons or else), then it's easier to manage the whole list of items (batch moving, exporting, deleting, etc).

**Reply by kylecorry31** [reply]
Hi, thank you for the suggestions. Here's a new issue for exploring the tree structure layout: https://github.com/kylecorry31/Trail-Sense/issues/2308

And the multiple selection of list items will be part of this issue once implemented: https://github.com/kylecorry31/Trail-Sense/issues/1151

**Comment by astewart77** [comment]
Is it possible to go directly from beacon navigation or the current location to the containing photo map, without knowing it's name? I have a trail that winds through the adjacent corners of several USGS pdfs. I can see the beacon if I open the correct photo map, but it's difficult to pick the right one from a long list.

**Reply by kylecorry31** [reply]
Hi, in the next release I'm adding a quick action to open the active photo map. That will be the map that you are currently on (choosing the one that's most zoomed in / with your location closest to the center): #2304

I'll experiment with modifying that behavior when you are navigating to a beacon to try to choose a map that contains both your location and the beacon (within reason - wouldn't want to choose a map of the larger region/country if no map contains both): #2376

**Comment by HidingInEmptySpace** [comment]
# New tool Suggestion: Hydration tracker

### Track how much water you drank.

Water ist the most valuable asset of the hiker, survivalist or any regular person. When planed without care, you may face serious problems to solve in the wilderness.

This tool could be enhanced by many additional features, to solve a variety of problems that are more or less necessary. For example:
- how much water is required for a certain path, maybe based on temperature, height or so
- tracking of the remaining water stock
      - reminder to replenish water when near certain position marked as watersource
      - put fellow hikers into account that drink from the same stock (maybe pairing feature ? Allthough simple workaround would be to carry your own)
- reminder to drink water

I suggest reviewing the variety of water-tracking-apps to include your favourite GUI features :-)

**Reply by kylecorry31** [reply]
Thank you for the suggestion, I will plan to add this tool: https://github.com/kylecorry31/Trail-Sense/issues/2412

**Comment by HidingInEmptySpace** [comment]
# new feature Suggestion: low battery notification and handling

The default low-battery-notifications of your phones are often not noticable enough for a survival-scenario when you rely on certain tools.
Therefore an
### additional notification
should appear to frequently inform you about the current state of the battery. Idealy - the more it reaches 0% the noticable it gets.

**Reply by kylecorry31** [reply]
Thank you, here's an issue for the low battery alerts: https://github.com/kylecorry31/Trail-Sense/issues/2413

As for the power saving mode - there's a option in Settings > Battery > Automatic power saver which activates the Trail Sense power saver mode when the system reports low power (typically when the system enters power saver mode). Unfortunately Trail Sense can't control the system power saver.

**Comment by HidingInEmptySpace** [comment]
# new feature Suggestion: Copy-Sheet

Containing the most valuable information gathered. When too low on battery, it should be displayed with extra big font, giving the user the last opportunity to copy important stuff onto paper by hand before you run out of energy or decide to shut down your device to use spare energy later.

Valuable information could be:
- Past notes saved in the app
- Directions & distance to important Points
- Distance & Time stamp of last thunderstorm
- General Knowledge and 'rules of thumb' on how to determine current direction, time and so on

**Reply by kylecorry31** [reply]
Great idea, here's a new issue to track it: https://github.com/kylecorry31/Trail-Sense/issues/2415

**Comment by vkuehn** [comment]
There are a lot tools in trail sense. I wonder if an old phone could be used with another trail sense to communicate. E.g. to publish sensor data from outside a shelter like the camera or mice!?

**Reply by kylecorry31** [reply]
Thank you for the suggestion. Unfortunately I don't have plans to have Trail Sense be used to stream sensor data/camera feeds between devices. The closest planned feature is the Comms plugin which will allow sending some data between devices (more of an offline file share/messaging though): #1540

**Reply by astewart77** [reply]
Phyphox - "Physical Phone Experiments" - https://phyphox.org/ might be of interest. It can be a web server for sensor data and associated experiments, which are user programmable. The latest beta has a couple of camera as brightness or hue sensor experiments. No video feeds.

**Reply by vkuehn** [reply]
Plugin's are not even offered. Is there a howto for it and how to handle/write plugins ?

**Reply by kylecorry31** [reply]
Plugins have not been implemented yet

**Reply by kylecorry31** [reply]
And I don't have plans to have 3rd party plugins at this point (if that's the reason you were asking)

**Comment by RuralAnemone** [comment]
would it be possible to add brown/pink/other color noise to the white noise section?

**Reply by kylecorry31** [reply]
Thank you for the suggestion. This feature is planned: #2506

**Comment by Craftefix** [comment]
I have a few things that would be nice:
- Smart watch support, not just as a standalone app, but also to get data from a gadget as in: #1534 
- File sharing over WiFi and Bluetooth, I personally use [LocalSend](https://github.com/localsend/localsend)
- Communication over Bluetooth and WiFi like in [NowiChat](https://github.com/cuadratico/NowiChat) (Only has Bluetooth), maybe even over Morse code utilizing the flash

**Reply by kylecorry31** [reply]
Thank you for the suggestions.

- Unfortunately I do not plan to add Android Wear support (see https://github.com/kylecorry31/Trail-Sense/issues/1467). Getting data from other devices will be tracked under #1540, but is unlikely to happen anytime soon.
- The Comms plugin on #1540 will cover both file share and communication over bluetooth/wifi.

**Comment by Craftefix** [comment]
An Option to download the Wikipedia articles mentioned in the Guide.
Thought that it cant be that difficult since they have an API.

Also choosing the language of the guide somewhere just temporary would be nice.

**Reply by jer194** [reply]
In Android 15 you can choose the language of the guide in 'System -> Languages -> App languages -> Trail Sense'
(if this is what you had asked for).

**Reply by Craftefix** [reply]
> In Android 15 you can choose the language of the guide in 'System -> Languages -> App languages -> Trail Sense'
> (if this is what you had asked for).

I ment like just one time directly when reading the guide,  if somethings unclear etc.

**Reply by kylecorry31** [reply]
Thank you for the suggestions:

1. Trail Sense does not have Internet permission so the wikipedia articles can't be downloaded via an API. You will need to download/print to PDF the page with your browser.
2. I recommend using the app language selector to change guide languages, due to Google Play app bundles, Trail Sense may not have the necessary content avaliable on device to switch languages. Therefore I unfortunately will not be implementing it. (https://developer.android.com/guide/app-bundle/configure-base#handling_language_changes)
